clientmodel/lib/src/Utopia/ClientModel.hs

```diff
@@ -197,6 +267,24 @@ instance FromJSON ProjectContentDirectory where
 instance ToJSON ProjectContentDirectory where
   toJSON = genericToJSON defaultOptions
 
+generateProjectContentsTree :: Int -> Gen ProjectContentsTree
+generateProjectContentsTree 0 = fmap ProjectContentsTreeFile arbitrary
+generateProjectContentsTree depth = oneof
+  [ fmap ProjectContentsTreeDirectory $ resize 3 $ generateProjectContentDirectory depth
+  , fmap ProjectContentsTreeFile arbitrary
+  ]
+
+generateProjectContentDirectory :: Int -> Gen ProjectContentDirectory
+generateProjectContentDirectory depth = do
+  fullPath <- arbitrary
+  children <- liftArbitrary $ generateProjectContentsTree (depth - 1)
+  let directory = Directory
+  pure ProjectContentDirectory{..}
```

Comment:
Likewise, can these also be moved please? It's hard to tell at first glance what code exists purely for tests vs what is the actual code

Reply:
As these are used in the `Arbitrary` instances they should probably stay where they are for previously mentioned reasons.